java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -195,11 +216,37 @@ public ColumnVector(DType type, long rows, Optional<Long> nullCount,
     for (int i = 0; i < nestedColumnVectors.size(); i++) {
```

Comment:
This constructor should be implemented in terms of the new `DataType` constructor to minimize the duplication of code and reduce maintenance.  For example, something like this:
```java
private static DataType toDataType(DType type) {
  assert !type.isDecimal() : "Cannot convert a decimal type without a specified scale";
  return new DataType(type);
}

[...]

public ColumnVector(DType type, long rows, Optional<Long> nullCount,
                    DeviceMemoryBuffer dataBuffer, DeviceMemoryBuffer validityBuffer,
                    DeviceMemoryBuffer offsetBuffer, List<NestedColumnVector> nestedColumnVectors) {
  this(toDataType(type), rows, nullCount, dataBuffer, validityBuffer, offsetBuffer, nestedColumnVectors);
}
```